This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fiesh <weiss@wsoptics.de>
Date: Tue, 22 Dec 2015 13:45:56 +0100
Subject: [PATCH] fix freeimage build

Add the FREEIMAGE_LIB define in order to keep the build process from
defining dllspec specifications. Remove -fPIC to fix a warning.

diff --git a/Makefile.fip b/Makefile.fip
index 1111111..2222222 100644
--- a/Makefile.fip
+++ b/Makefile.fip
@@ -17,7 +17,7 @@ MODULES = $(SRCS:.c=.o)
 MODULES := $(MODULES:.cpp=.o)
 CFLAGS ?= -O3 -fPIC -fexceptions -fvisibility=hidden -DNO_LCMS
 CFLAGS += $(INCLUDE)
-CXXFLAGS ?= -O3 -fPIC -fexceptions -fvisibility=hidden -Wno-ctor-dtor-privacy
+CXXFLAGS ?= -O3 -fPIC -fexceptions -fvisibility=hidden -Wno-ctor-dtor-privacy -DFREEIMAGE_LIB
 CXXFLAGS += $(INCLUDE)
 
 ifeq ($(shell sh -c 'uname -m 2>/dev/null || echo not'),x86_64)
From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Boris Pek <tehnick-8@mail.ru>
Date: Wed, 04 May 2016 18:43:16 +0300
Subject: [PATCH] fix build with GCC >= 6.x

diff --git a/Makefile.mingw b/Makefile.mingw
index 1111111..2222222 100644
--- a/Makefile.mingw
+++ b/Makefile.mingw
@@ -118,11 +118,8 @@ $(DISTDIR)/%: $(SRCDIR)/%
 $(STATICLIB): $(MODULES)
 	$(AR) rs $@ $(MODULES)
 
-$(IMPORTLIB) $(EXPORTLIB): $(MODULES) 
-	$(DLLTOOL) -e $(EXPORTLIB) -l $(IMPORTLIB) -D $(SHAREDLIB) $(DLLTOOLFLAGS) $(MODULES)
-
-$(SHAREDLIB): $(EXPORTLIB) $(RESOURCE)
-	$(LD) $(LDFLAGS) -o $@ $(EXPORTLIB) $(MODULES) $(RESOURCE) $(LIBRARIES)
+$(SHAREDLIB) $(IMPORTLIB): $(MODULES) $(RESOURCE)
+	$(LD) $(LDFLAGS) -o $@ $(MODULES) -Wl,--out-implib=$(IMPORTLIB) $(RESOURCE) $(LIBRARIES)
 
 $(DISTDIR):
 	$(MD) $(DISTDIR)
@@ -133,4 +130,4 @@ install:
 	$(CP) $(SHAREDLIB) $(INSTALLDIR)
 
 clean:
-	$(RM) -f core $(DISTDIR)/*.* $(MODULES) $(RESOURCE) $(STATICLIB) $(SHAREDLIB) $(IMPORTLIB) $(EXPORTLIB)
+	$(RM) -f core $(DISTDIR)/*.* $(MODULES) $(RESOURCE) $(STATICLIB) $(SHAREDLIB) $(IMPORTLIB)
